frontend/src/pages/class/[classId]/session/[sessionId]/join.tsx

```diff
@@ -215,10 +269,18 @@ const JoinSession = () => {
             studentPublicKey: JSON.stringify(await keyPair.exportPublicKey()),
             encryptedAuthenticationRequest: encodeBase64(
               await ephemeralKey.encryptString(
-                JSON.stringify({
-                  classId,
-                  idToken: authenticationContext.idToken,
-                } as StudentAuthenticationRequestContent),
+                JSON.stringify(
+                  (studentContext
```

Comment:
`studentContext` here is a proxy for "are we using anonymous auth". 
I'd strongly suggest using an appropriately-named variable to make this explicit.
Perhaps isSessionAnonymous ?

Reply:
Not just that, it is also used as the authentication context. Without this extra variable type inference does not seem to work..